### PR TITLE
Add a job to build the docs on CI

### PR DESCRIPTION
**Pull request by LucyJimenez**
xref https://github.com/airspeed-velocity/asv/issues/1063
On this PR create a job for docs

### CODE REVIEW

**Comment by datapythonista** [comment]
Did you check if the docs are being built correctly?

**Comment by LucyJimenez** [comment]
> Did you check if the docs are being built correctly?

Yes, now it’s possible to create the HTML. I’m working on fixing the problems, and I’ll be listing the issues here.

- `asv/docs/source/manindex.rst:document isn't included in any toctree`
- `asv/docs/source/benchmarks.rst:148:py:obj reference target not found: time.clock`

**Comment by LucyJimenez** [comment]
@datapythonista The PR is ready for your review; thanks!

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -47,3 +47,18 @@ jobs:
 
       - name: Check code with flake8
         run: flake8 .
+
+  docs:
+    name: builds the documentation using Sphinx
```
Comment (datapythonista) [inline .github/workflows/ci.yml:52]: I think you can delete this row. The default `docs` is easier to read in my opinion.

File: docs/source/reference.rst [hunk 2]
```diff
@@ -3,6 +3,7 @@ Reference
 
 .. toctree::
 
+   manindex.rst
```
Comment (datapythonista) [inline docs/source/reference.rst:6]: This seems to be a duplicate of reference, just remove the page instead of including it here.

**Comment by LucyJimenez** [comment]
@datapythonista, I included the last comments. The PR is ready for review. Thanks a lot!

File: .github/workflows/ci.yml [hunk 3]
```diff
@@ -46,6 +46,20 @@ jobs:
 
       - name: flake8
         run: flake8 .
-
+
```
Comment (datapythonista) [inline .github/workflows/ci.yml:49]: You should review the diffs when you're done, and things like this shouldn't be introduced by mistake.

**Review by datapythonista — COMMENTED** [review]
If having an `xref` in the description correct? Or should we have a close and let this automatically close the issue?

**Comment by datapythonista** [comment]
Thanks @LucyJimenez

**Pull request merged**